File: project/player_init.lua
local skynet = require "skynet"
local socket = require "skynet.socket"
local mysql = require "mysql"

Player_init = {}

function Player_init.init(cID)
	socket.write(cID, "ok")
	skynet.sleep(5)
	socket.write(cID, "ok") 
    local db = mysql.connect()
    local name = socket.read(cID)
    socket.write(cID, "ok")
    local job = socket.read(cID)
    socket.write(cID, "ok")
    local res = db:query(string.format("select * from %s where name = \'%s\'", job, name))
	mysql.dump(res)
    
    socket.write(cID, res[1]["blood"])
    local str = socket.read(cID)
    while str ~= "ok" do
    end

    socket.write(cID, res[1]["attack"])
    local str = socket.read(cID)
    while str ~= "ok" do
    end

    socket.write(cID, res[1]["defense"])
    local str = socket.read(cID)
    while str ~= "ok" do
    end

    if res[1]["arm"] == nil then
        socket.write(cID, "NULL")
    else
        socket.write(cID, res[1]["arm"])
    end
    str = socket.read(cID)
    while str ~= "ok" do
    end

    if res[1]["armor"] == nil then
        socket.write(cID, "NULL")
    else
        socket.write(cID, res[1]["armor"])
    end
	res = db:query(string.format("update %s set cID = %d where name = \'%s\'", job, cID, name))
	mysql.dump(res)
	db:disconnect()
	return name, job
end

return Player_init

File: project/player_init.cpp
#include "player_init.h"

using namespace std;

int player_init(Character *cha, int sockfd, string name, string job, vector<Things_str*>&package)
{
    string str;
    int n = 0;
    char buf[MAXLINE];

    pth_simble = 1;
    str = "player_init";
    write(sockfd, str.c_str(), str.size());
    pthread_mutex_lock(&lock);
    
    n = read(sockfd, buf, MAXLINE);
    if(n == 0)
        print_disconnect(sockfd);
    str = buf;
    while(str != "ok");

    write(sockfd, name.c_str(), name.size());
    memset(buf, 0, MAXLINE);
    n = read(sockfd, buf, MAXLINE);
    if(n == 0)
        print_disconnect(sockfd);
    str = buf;
    while(str != "ok");

    write(sockfd, job.c_str(), job.size());
    memset(buf, 0, MAXLINE);
    n = read(sockfd, buf, MAXLINE);
    if(n == 0)
        print_disconnect(sockfd);
    str = buf;
    while(str != "ok");

    int blood, attack, defense;

    memset(buf, 0, MAXLINE);
    n = read(sockfd, buf, MAXLINE);
    if(n == 0)
        print_disconnect(sockfd);
    blood = atoi(buf);
    str = "ok";
    write(sockfd, str.c_str(), str.size());

    memset(buf, 0, MAXLINE);
    n = read(sockfd, buf, MAXLINE);
    if(n == 0)
        print_disconnect(sockfd);
    attack = atoi(buf);
    str = "ok";
    write(sockfd, str.c_str(), str.size());

    memset(buf, 0, MAXLINE);
    n = read(sockfd, buf, MAXLINE);
    if(n == 0)
        print_disconnect(sockfd);
    defense = atoi(buf);
    str = "ok";
    write(sockfd, str.c_str(), str.size());

	int ret;
    memset(buf, 0, MAXLINE);
    n = read(sockfd, buf, MAXLINE);
    if(n == 0)
        print_disconnect(sockfd);
    str = buf;
	if(str != "NULL")
	{
		ret = package_find(package, str);
		if(ret != -1)
		{
			cha->Puton_arm(package[ret]->things);
			package[ret]->things->init(cha);
		}
	}
	str = "ok";
    write(sockfd, str.c_str(), str.size());

    memset(buf, 0, MAXLINE);
    n = read(sockfd, buf, MAXLINE);
    if(n == 0)
        print_disconnect(sockfd);
    str = buf;
	if(str != "NULL")
	{
		ret = package_find(package, str); 
		if(ret != -1)
		{
			cha->Puton_armor(package[ret]->things);
			package[ret]->things->init(cha);
		}
	}
	
	int m_blood = cha->Getblood();
	int m_attack = cha->Getattack();
	int m_defense = cha->Getdefense();
	if(m_blood != blood || m_attack != attack || m_defense != defense)
		cout << "属性与服务端数据不同！" << endl;
    
    pth_simble = 0;
    pthread_mutex_unlock(&lock);
	pthread_cond_signal(&pth_do);
    return 1;
}
